crates/ruff_shrinking/src/main.rs

```diff
@@ -269,12 +271,61 @@ impl Strategy for StrategyRemoveNewline {
             .collect();
         let iter = newline_positions.into_iter().map(move |newline_position| {
             // trim to remove the indentation
-            input[..newline_position].to_string() + input[newline_position + 1..].trim_start()
+            input[..newline_position].to_string()
+                + input[newline_position + '\n'.len_utf8()..].trim_start()
         });
         Ok(Box::new(iter))
     }
 }
 
+/// Try removing each python token. This is really slow and runs at the end
+struct StrategyRemoveToken;
+
+impl Strategy for StrategyRemoveToken {
+    fn name(&self) -> &'static str {
+        "remove token"
+    }
+
+    fn candidates<'a>(
+        &self,
+        input: &'a str,
+        _ast: &'a Suite,
+    ) -> Result<Box<dyn ExactSizeStringIter + 'a>> {
+        let token_ranges: Vec<_> = ruff_rustpython::tokenize(input)
+            .into_iter()
+            .map(|token| token.unwrap())
```

Comment:
Use `flatten` to not throw on invalid syntax OR should this bail if the input syntax is invalid?

Reply:
added a comment because we know we have valid python here (we get the AST)